sphinx/domains/cpp/_parser.py

```diff
@@ -1788,7 +1788,7 @@ def _parse_template_parameter(self) -> ASTTemplateParam:
             except DefinitionError as eNonType:
                 self.pos = pos
                 header = "Error when parsing template parameter."
-                errs = []
+                errs: list[tuple[DefinitionError, str]] = []
                 errs.append(
```

Comment:
Here, it'd be better to just write something like `errs = [...]` without those un-necessary append calls.